### ISSUE TITLE
Cannot add click-params to project

### ISSUE THREAD

**Issue by Tatsh** [python-poetry/poetry#8434]
- **Poetry version**: 1.6.1
- **Python version**: 3.11.5
- **OS version and name**: Gentoo
- **pyproject.toml**: n/a because this happens with a bare minimum pyproject.toml file

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

Trying to add click-params to any project results in this error: *Can't instantiate abstract class IsolatedEnv with abstract methods make_extra_environ, python_executable*.

Full output:

```
Loading configuration file /home/tatsh/.config/pypoetry/config.toml
Using virtualenv: /home/tatsh/.cache/pypoetry/virtualenvs/newproj-0wsIROfq-py3.11
[keyring.backend] Loading KWallet
[keyring.backend] Loading SecretService
[keyring.backend] Loading Windows
[keyring.backend] Loading chainer
[keyring.backend] Loading libsecret
[keyring.backend] Loading macOS
Creating new session for pypi.org
Source (PyPI): 7 packages found for click-params *
Using version ^0.4.1 for click-params

Updating dependencies
Resolving dependencies...
   1: fact: newproj is 0.1.0
   1: derived: newproj
   1: fact: newproj depends on click-params (^0.4.1)
   1: selecting newproj (0.1.0)
   1: derived: click-params (>=0.4.1,<0.5.0)
Source (PyPI): 1 packages found for click-params >=0.4.1,<0.5.0
   1: fact: click-params (0.4.1) depends on click (>=7.0,<9.0)
   1: fact: click-params (0.4.1) depends on validators (>=0.20,<0.21)
   1: selecting click-params (0.4.1)
   1: derived: validators (>=0.20,<0.21)
   1: derived: click (>=7.0,<9.0)
Source (PyPI): 1 packages found for validators >=0.20,<0.21
Source (PyPI): 17 packages found for click >=7.0,<9.0
   1: fact: validators (0.20.0) depends on decorator (>=3.4.0)
   1: selecting validators (0.20.0)
   1: derived: decorator (>=3.4.0)
Source (PyPI): 28 packages found for decorator >=3.4.0
   1: selecting decorator (5.1.1)
   1: fact: click (8.1.7) depends on colorama (*)
   1: selecting click (8.1.7)
   1: derived: colorama
Source (PyPI): 43 packages found for colorama *
   1: selecting colorama (0.4.6)
   1: Version solving took 0.043 seconds.
   1: Tried 1 solutions.

Finding the necessary packages for the current system
Source (PyPI): 1 packages found for click-params >=0.4.1,<0.5.0
Source (PyPI): 1 packages found for validators >=0.20,<0.21
Source (PyPI): 1 packages found for click >=7.0,<9.0
Source (PyPI): 1 packages found for decorator >=3.4.0

Package operations: 2 installs, 0 updates, 0 removals, 2 skipped

  • Installing validators (0.20.0)

  Stack trace:

  8  /usr/lib/python3.11/site-packages/poetry/installation/executor.py:286 in _execute_operation
      284│ 
      285│             try:
    → 286│                 result = self._do_execute_operation(operation)
      287│             except EnvCommandError as e:
      288│                 if e.e.returncode == -2:

  7  /usr/lib/python3.11/site-packages/poetry/installation/executor.py:386 in _do_execute_operation
      384│             return 0
      385│ 
    → 386│         result: int = getattr(self, f"_execute_{method}")(operation)
      387│ 
      388│         if result != 0:

  6  /usr/lib/python3.11/site-packages/poetry/installation/executor.py:511 in _execute_install
      509│ 
      510│     def _execute_install(self, operation: Install | Update) -> int:
    → 511│         status_code = self._install(operation)
      512│ 
      513│         self._save_url_reference(operation)

  5  /usr/lib/python3.11/site-packages/poetry/installation/executor.py:549 in _install
      547│             archive = self._download_link(operation, Link(package.source_url))
      548│         else:
    → 549│             archive = self._download(operation)
      550│ 
      551│         operation_message = self.get_operation_message(operation)

  4  /usr/lib/python3.11/site-packages/poetry/installation/executor.py:753 in _download
      751│             self._yanked_warnings.append(message)
      752│ 
    → 753│         return self._download_link(operation, link)
      754│ 
      755│     def _download_link(self, operation: Install | Update, link: Link) -> Path:

  3  /usr/lib/python3.11/site-packages/poetry/installation/executor.py:800 in _download_link
      798│             self._write(operation, message)
      799│ 
    → 800│             archive = self._chef.prepare(archive, output_dir=output_dir)
      801│ 
      802│         # Use the original archive to provide the correct hash.

  2  /usr/lib/python3.11/site-packages/poetry/installation/chef.py:101 in prepare
       99│             return self._prepare(archive, destination=destination, editable=editable)
      100│ 
    → 101│         return self._prepare_sdist(archive, destination=output_dir)
      102│ 
      103│     def _prepare(

  1  /usr/lib/python3.11/site-packages/poetry/installation/chef.py:185 in _prepare_sdist
      183│             destination.mkdir(parents=True, exist_ok=True)
      184│ 
    → 185│             return self._prepare(
      186│                 sdist_dir,
      187│                 destination,

  TypeError

  Can't instantiate abstract class IsolatedEnv with abstract methods make_extra_environ, python_executable

  at /usr/lib/python3.11/site-packages/poetry/installation/chef.py:109 in _prepare
      105│     ) -> Path:
      106│         from subprocess import CalledProcessError
      107│ 
      108│         with ephemeral_environment(self._env.python) as venv:
    → 109│             env = IsolatedEnv(venv, self._pool)
      110│             builder = ProjectBuilder(
      111│                 directory,
      112│                 python_executable=env.executable,
      113│                 scripts_dir=env.scripts_dir,

```

**Comment by dimbleby** [comment]
you have installed poetry incorrectly - you are finding a newer version of `build` than is allowed by poetry's requirements.

please close

**Comment by oktavlachs** [comment]
I found a solution for Arch, so in case someone is looking for a workaround while waiting for the next Poetry release, here you go:

The reason for this error is that the "build" dependency has been bumped to 1.0.3 lately in the Arch repos and that is not supported by the current Poetry release.

There is a little cli tool called "downgrade" that you can find in the official Arch repos.
Install it, run sudo downgrade python-build , set the version to 0.10.0 and et voilà. 👍 Hope that helps.

**Comment by moson-mo** [comment]
> There is a little cli tool called "downgrade" that you can find in the official Arch repos. Install it, run `sudo downgrade python-build` , set the version to 0.10.0 and et voilà. 👍 Hope that helps.

In the meanwhile there is 1.6.1-**2** in the Arch repos. That should be fixing it.

**Comment by MarcelWilson** [comment]
> you have installed poetry incorrectly - you are finding a newer version of `build` than is allowed by poetry's requirements.

@dimbleby Can you ELI5?  is `build` in this case a python package?  And if so, why would poetry not support installing a specific version of a package?

**Comment by sirex** [comment]
On Manjaro, following command fix the issue for me:

    sudo pacman -U file:///var/cache/pacman/pkg/python-build-0.10.0-4-any.pkg.tar.zst

This downgraded python-build from `1.0.1-1` to `0.10.0-4`.
